### PR TITLE
Add v1.5.1

### PR DESCRIPTION
**Pull request by 0-vortex**
Combined pr from all the forks. The initial project has been forked a few times with a few minor changes:
- base class test
- update mechanism
- latest version added

This is mainly based on the work of imdario and signalfx with some thorough testing on centos 6.7 x64. Not all the contributors were mentioned so far.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -10,23 +10,33 @@
...
+# Dario Castañé <i@dario.im>
+# TED Vortex    <ted.vortex@gmail.com>
 #
+
 class golang (
-  $version      = '1.1.2',
-  $workspace    = '/vagrant',
-  $arch         = 'linux-amd64',
-  $download_dir = '/usr/local/src',
+  $version      = "1.5.1",
```
Comment (dcoxall) [inline manifests/init.pp]: The version here differs to that in the README

**Comment by dcoxall** [comment]
What is this go binary updater? I'm not familiar with it.

File: templates/golang.sh.erb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 GOPATH="<%= @workspace %>"
 PATH="$GOPATH/bin:/usr/local/go/bin:$PATH"
-export GOPATH PATH
+GO15VENDOREXPERIMENT=1
```
Comment (dcoxall) [inline templates/golang.sh.erb]: Not sure if it's worth making this configurable seeing as it is just an experiment

**Comment by dcoxall** [comment]
> base class test

I misunderstand as I was soo excited to see a test for this haha

**Comment by dcoxall** [comment]
I'm going to give this a little test later this week but really awesome that you all took the time to contribute you time and changes upstream! You Rock! :tada: 

I've left a few comments but they're open to discussion. Given the nature of some of the history it would be great if (once all the changes are complete) we could squash the changes.

Again, thanks for the contribution!

**Comment by vrtxf** [comment]
Hi, I'll add a few commits to this following your guidelines but I am not familiar with the go language or best practices involving it. Will need some feedback if you can check back in a couple of days or so !

**Comment by imdario** [comment]
:+1: Cool PR. When it is done I'll close my repo, just to have a canonical one.

**Pull request closed without merging**